[feat] win/lose handling

RadicalV opened the pull request.

Commit:
[feat] win/lose handling

DovydasSp commented on client/src/boundary/model/BoundaryGame.ts:
```diff
@@ -2,12 +2,12 @@ import BoundaryBoard from "./BoundaryBoard";
 export default class BoundaryGame {
   readonly id: string;
-  readonly active: boolean;
+  readonly state: string;
```
vietoj string siūlau naudoti enum arba `state: 'IN_PROGRESS' | 'SOME_OTHER' | 'ANOTHER_ONE';`

DovydasSp replied:
'IN_PROGRESS' | 'SOME_OTHER' | 'ANOTHER_ONE' šitą galima išsikelti į kokį global `type`

DovydasSp commented on client/src/ui/containers/useGameStats.ts:
```diff
@@ -2,14 +2,20 @@ import { gameStatsController } from "config";
     gameStatsController().getGameStats(gameId).subscribe(setGameStats);
-  }, [board, gameId]);
+    onStatsUpdate(gameStats.shipsDestroyed);
+    console.log("use effect");
```
nepalik tokių

DovydasSp commented on client/src/ui/components/GameOvermodal.test.tsx:
```diff
@@ -0,0 +1,44 @@
+import { IN_PROGRESS, LOST, WON } from "utils/Constants";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+describe("Game over modal", () => {
```
nėra testo kai gameState = IN_PROGRESS

Commit:
string -> enum swap, GameOverModal test

changed game state from string into a enum. Wrote a game over modal test when game state is IN_PROGRESS
